plugins/lua/confirm.lua

```diff
@@ -1,4 +1,26 @@
 local _ENV = mkmodule('plugins.confirm')
+local GLOBAL_KEY = 'confirm'
+local CONFIRM_CONFIG_FILE = 'dfhack-config/confirm.json'
+local json = require('json')
+
+loaded = loaded or false
+
+if dfhack.onStateChange then
+    dfhack.onStateChange[GLOBAL_KEY] = function(_)
+        if loaded then
+            return
+        end
+
+        local ok, config = pcall(json.decode_file, CONFIRM_CONFIG_FILE)
+        local conf_data = ok and config or get_conf_data()
+        if ok then
+            for _, c in ipairs(conf_data) do
+                set_conf_state(c.id, c.enabled)
+            end
+        end
+        loaded = true
+    end
+end
```

Comment:
splitting the save/restore logic across files seems like asking for trouble down the line. could you move this logic into gui/confirm?

Reply:
note that you have to make `gui/confirm` a `--@ module=true`

Reply:
I did initially try doing this but it made the confirm options pop up as soon as the game loaded and I wasn't sure the best way to stop that from happening

Reply:
As soon as I add `--@ module=true`:

![image](https://user-images.githubusercontent.com/2104555/218611873-8657b2b3-4bd3-40a2-ad85-6f43c2de8fad.png)

Reply:
you can't just *say* you're a module. you also have to *act* like a module : )

add:
```lua
if dfhack_flags.module then
    return
end
```

before the script has any side effects (like bringing up a window)

Reply:
e.g. https://github.com/DFHack/scripts/blob/b13f351b22de50210b9b71d34806ebc2b84351ca/gui/quickfort.lua#L677

Reply:
Haha.. sorry. I'm still covid dumb. That's what I missed. Thanks.